### PR TITLE
Issue #1788 Fix commit message validation regex

### LINKED ISSUE #1788: Fix commit message validation regex
[Commit message validation](https://docs.openshift.org/latest/minishift/contributing/developing.html#commit-message-requirements) seems to be using the wrong regex here 

https://github.com/minishift/minishift/blob/e99350316a8a15a7f489f15aeec44b919fd78c71/Makefile#L233

I was expecting my PR https://github.com/minishift/minishift/pull/1786  with message `Must fail Issue #1785 Validate commit message as part of CI builds` should fail but it is not.

### PR DESCRIPTION
**Pull request by budhrg**
Fix #1788

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -230,4 +230,4 @@ fmtcheck:
 
 .PHONY: validate_commits
 validate_commits: $(GOPATH)/bin/git-validation
-	git-validation -q -run short-subject,message_regexp='(Issue #[0-9]+ .*|cut v[0-9]+.[0-9]+.[0-9]+)' -range $(START_COMMIT_MESSAGE_VALIDATION)...
+	git-validation -q -run short-subject,message_regexp='(^Issue #[0-9]+ .*|^cut v[0-9]+.[0-9]+.[0-9]+)' -range $(START_COMMIT_MESSAGE_VALIDATION)...
```
Comment (robin-owen) [inline Makefile]: Shouldn't `^(Issue #[0-9]+ .*|cut v[0-9]+.[0-9]+.[0-9]+)` be equivalent?

Comment (gbraad) [reply Makefile]: https://regex101.com/r/Iv69Jr/1

Comment (gbraad) [reply Makefile]: `^(Issue #[0-9]+ .*|cut v[0-9]+.[0-9]+.[0-9]+$)`

File: Makefile [hunk 2]
```diff
@@ -232,4 +232,4 @@ fmtcheck:
 
 .PHONY: validate_commits
 validate_commits: $(GOPATH)/bin/git-validation
-	git-validation -q -run short-subject,message_regexp='(Issue #[0-9]+ .*|cut v[0-9]+.[0-9]+.[0-9]+)' -range $(START_COMMIT_MESSAGE_VALIDATION)...
+	git-validation -q -run short-subject,message_regexp="^(Issue #[0-9]+ .*|cut v[0-9]+.[0-9]+.[0-9]+)$$" -range $(START_COMMIT_MESSAGE_VALIDATION)...
```
Comment (gbraad) [inline Makefile]: Why was this changed to a double-quotes from single-quotes? This way you will have an issue with shell interpretation/evaluation of eg. variables (`$`)
